### PR TITLE
:sparkles: Switch

### PR DESCRIPTION
**Pull request by KipandaJr**

### CODE REVIEW

File: src/lib/components/switch/switch.types.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export type LabelElementProps = Omit<
+  LabelHTMLAttributes<HTMLInputElement>,
+  'color' | 'width' | 'height' | 'size'
+>;
+export interface CheckedButtonProps extends CheckedButtonElementProps {
+  name: string;
+  size: string;
+  hideLabel?: boolean;
+  options: ReadonlyArray<string>;
+  initialValue: boolean;
```
Comment (git-marcopitra) [inline src/lib/components/switch/switch.types.ts:17]: I it is a normal input, for sure, it has a default value, just use it

File: src/lib/components/switch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  CheckedButtonElementProps,
+  CheckedButtonProps,
+  LabelElementProps,
+} from './switch.types';
+
+export const SwitchButton: FC<PropsWithChildren<CheckedButtonProps>> = ({
+  size,
+  hideLabel,
+  options,
+  initialValue,
```
Comment (git-marcopitra) [inline src/lib/components/switch/index.tsx:17]: replace for `defaultValue`

File: src/lib/components/switch/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          borderRadius={`calc(${size}/2.3)`}
+          nBefore={{
+            content: '""',
+            position: 'absolute',
+            width: `calc(${size}/2.3)`,
+            height: `calc(${size}/2.3)`,
+            borderRadius: '50%',
+            bottom: `calc(${size}/30)`,
+            left: `calc(${size}/30)`,
+            bg: dark ? 'foreground' : 'background',
```
Comment (git-marcopitra) [inline src/lib/components/switch/index.tsx:80]: avoid handle the color using `dark`, only if it's unavoidable

Comment (git-marcopitra) [reply src/lib/components/switch/index.tsx:80]: In this case, it can keep, but just be care with the next cases

**Review by git-marcopitra — CHANGES_REQUESTED** [review]

**Review by okusindja — COMMENTED** [review]
Done ✅

**Pull request merged**